### ISSUE TITLE
CheckStringIndexValueCodeRefactoringProvider crash

### ISSUE THREAD

**Issue by Therzok** [mono/monodevelop#6653]
```
ERROR [2018-11-24 03:33:02Z]: Roslyn error: Extension_Exception CheckStringIndexValueCodeRefactoringProvider : Specified argument was out of the range of valid values.
Parameter name: index
  at Microsoft.CodeAnalysis.SeparatedSyntaxList`1[TNode].get_Item (System.Int32 index) [0x00039] in /_/src/Compilers/Core/Portable/Syntax/SeparatedSyntaxList.cs:103 
  at Microsoft.CodeAnalysis.SeparatedSyntaxList`1[TNode].First () [0x00000] in /_/src/Compilers/Core/Portable/Syntax/SeparatedSyntaxList.cs:182 
  at RefactoringEssentials.CSharp.CodeRefactorings.CheckStringIndexValueCodeRefactoringProvider.ComputeRefactoringsAsync (Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringContext context) [0x0028d] in /Users/vsts/agent/2.142.1/work/1/s/monodevelop/main/external/RefactoringEssentials/RefactoringEssentials/CSharp/CodeRefactorings/Synced/CheckStringIndexValueCodeRefactoringProvider.cs:44 
  at Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringService.GetRefactoringFromProviderAsync (Microsoft.CodeAnalysis.Document document, Microsoft.CodeAnalysis.Text.TextSpan state, Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringProvider provider, Microsoft.CodeAnalysis.Extensions.IExtensionManager extensionManager, System.Threading.CancellationToken cancellationToken) [0x000a0] in /_/src/Features/Core/Portable/CodeRefactorings/CodeRefactoringService.cs:143 
ERROR [2018-11-24 03:33:10Z]: Roslyn reported an exception to the user
System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values.
Parameter name: index
  at Microsoft.CodeAnalysis.SeparatedSyntaxList`1[TNode].get_Item (System.Int32 index) [0x00039] in /_/src/Compilers/Core/Portable/Syntax/SeparatedSyntaxList.cs:103 
  at Microsoft.CodeAnalysis.SeparatedSyntaxList`1[TNode].First () [0x00000] in /_/src/Compilers/Core/Portable/Syntax/SeparatedSyntaxList.cs:182 
  at RefactoringEssentials.CSharp.CodeRefactorings.CheckStringIndexValueCodeRefactoringProvider.ComputeRefactoringsAsync (Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringContext context) [0x0028d] in /Users/vsts/agent/2.142.1/work/1/s/monodevelop/main/external/RefactoringEssentials/RefactoringEssentials/CSharp/CodeRefactorings/Synced/CheckStringIndexValueCodeRefactoringProvider.cs:44 
  at Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringService.GetRefactoringFromProviderAsync (Microsoft.CodeAnalysis.Document document, Microsoft.CodeAnalysis.Text.TextSpan state, Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringProvider provider, Microsoft.CodeAnalysis.Extensions.IExtensionManager extensionManager, System.Threading.CancellationToken cancellationToken) [0x000a0] in /_/src/Features/Core/Portable/CodeRefactorings/CodeRefactoringService.cs:143 
```

> VS bug [#734935](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/734935)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-3] Comment by **xamarinc**



Fixed in version 8.0.0.1004 (master)



Author: Marius Ungureanu  
 Commit: 270b19bec12bf43c1fc2e322d58d1dde7b69ce10 (mono/RefactoringEssentials)  
 Included in Commit: a68f4f65f8d404402f7878ca5ebbc0c401111929 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-5] Comment by **xamarinc**



Fixed in version 8.0.0.1027 (release-8.0)



Author: Marius Ungureanu  
 Commit: 270b19bec12bf43c1fc2e322d58d1dde7b69ce10 (mono/RefactoringEssentials)  
 Included in Commit: 37c73ab967af232ed5ed087ecf57a6dc190e51cc (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-6] Comment by **xamarinc**



Fixed in version 8.0.0.1738 (release-8.0)



Author: Marius Ungureanu  
 Commit: 270b19bec12bf43c1fc2e322d58d1dde7b69ce10 (mono/RefactoringEssentials)  
 Included in Commit: 2dd1f9e4292065568730b377c049435601eea319 (xamarin/monodevelop)
